### ISSUE TITLE
Don't Upgrade If Upgrade Version Doesn't Work With Current Operating System

### ISSUE THREAD

**Issue by Miserlou** [docker/for-mac#1483]
If I'm running on an old OSX, please don't allow the software to update to a version which is incompatible with my OS. Currently you install and warn rather than checking first. This is going to impact a lot of people if you don't get ahead of this one.

**Comment by avsm** [comment]
Thanks for the bug report! Which version of macOS are you upgrading to that is not compatible with Docker for Mac?  We generally support all the latest versions, although of course developer and beta seeds may exhibit errors.

**Comment by Miserlou** [comment]
I'm filing this bug report _before_ the damage occurs because I just saw how your upgrade happened. It doesn't look like you're checking operating system compatibility before the upgrade occurs. I just upgraded my version and I got an imminent deprecation warning, but I only got the warning _after_ upgrading, not before, meaning you're most likely doing this in the wrong order and allowing people to upgrade to an incompatible version.

**Comment by avsm** [comment]
Thanks for the clarification. I've filed an internal bug report with the feature enhancement you've suggested.  We haven't yet actually deprecated any versions of macOS yet, but it is likely to happen with macOS 10.10 once another major revision of macOS is released, so this is an important aspect to consider.

I'll close this bug for now as there's not much point holding it open -- we've got the internal bug as a reminder to implement this feature before deprecating a currently supported macOS version.  Thanks again for taking the time to report it!
